shell/platform/linux/fl_key_event_channel.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_KEY_EVENT_CHANNEL_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_KEY_EVENT_CHANNEL_H_
+
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_binary_messenger.h"
+
+G_BEGIN_DECLS
+
+/**
+ * FlKeyEventType:
+ * @FL_KEY_EVENT_TYPE_DOWN: Key is pressed.
+ * @FL_KEY_EVENT_TYPE_UP: Key is released.
+ *
+ * Types of key events.
+ */
+typedef enum {
+  FL_KEY_EVENT_TYPE_UP,
+  FL_KEY_EVENT_TYPE_DOWN,
+} FlKeyEventType;
+
+G_DECLARE_FINAL_TYPE(FlKeyEventChannel,
```

Comment:
s/Channel/Plugin/ for this class name, since it's not a channel. (It's not 100% a plugin either, but that's the name we use for this helper on at least some other platforms, and this is much more like a plugin than it is a channel.)